test/AutoRest.TestServerLowLevel.Tests/LowLevel/Generation/ModelWriterTests.cs

```diff
@@ -60,6 +60,192 @@ public InputModel(string requiredString,int requiredInt)
 }
 }
 }
+");
+        }
+
+        [Test]
+        public void PrimitiveProperties()
+        {
+            // refer to the original CADL file: https://github.com/Azure/cadl-ranch/blob/main/packages/cadl-ranch-specs/http/models/primitive-properties/main.cadl
+            var model = new ModelTypeProvider(
+                new InputModelType("PrimitivePropertyModel", "Cadl.TestServer.PrimitiveProperties", "public",
+                    new List<InputModelProperty>{
+                        new InputModelProperty("requiredString", "requiredString", "", InputPrimitiveType.String, true, false, false),
+                        new InputModelProperty("requiredInt", "requiredInt", "", InputPrimitiveType.Int32, true, false, false),
+                        new InputModelProperty("requiredLong", "requiredLong", "", InputPrimitiveType.Int64, true, false, false),
+                        new InputModelProperty("requiredSafeInt", "requiredSafeInt", "", InputPrimitiveType.Int64, true, false, false),
+                        new InputModelProperty("requiredFloat", "requiredFloat", "", InputPrimitiveType.Float32, true, false, false),
+                        new InputModelProperty("requiredDouble", "requiredDouble", "", InputPrimitiveType.Float64, true, false, false),
+                        new InputModelProperty("requiredBodyDateTime", "requiredBodyDateTime", "Illustrate a zonedDateTime body parameter, serialized as (https://datatracker.ietf.org/doc/html/rfc3339)", InputPrimitiveType.DateTimeISO8601, true, false, false),
+                        new InputModelProperty("requiredDuration", "requiredDuration", "", InputPrimitiveType.DurationISO8601, true, false, false),
+                        new InputModelProperty("requiredBoolean", "requiredBoolean", "", InputPrimitiveType.Boolean, true, false, false)
+                    },
+                    null, null, null),
+                new TypeFactory(null),
+                "test",
+                null);
+
+            var codeWriter = new CodeWriter();
+            LowLevelModelWriter.WriteType(codeWriter, model);
+            var codes = codeWriter.ToString();
+
+            Assert.AreEqual(codes, @"// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Can we use an expected.cs to store the expected result? In a string, it may hard to maintain, and update.

Reply:
@chunyu3 The `*.cs` file will be compiled into byte codes, so they don't exist at runtime. We could use`*.cs.txt` but that requires we add them into the target DLL which might be overkill. I will strip the multi-line texts into a separate source.